legacy/firmware/trezor.c

```diff
@@ -117,11 +118,20 @@ void check_lock_screen(void) {
   }
 }
 
+void trezor_set_busy(uint32_t length_ms) {
+  system_millis_busy_start = timer_ms();
+  system_millis_busy_length = length_ms;
+}
+
+bool trezor_is_busy(void) {
+  return (uint32_t)(timer_ms() - system_millis_busy_start) <
```

Comment:
IIUC, `timer_ms()`'s return value and `system_millis_busy_start` are both `uint32_t` - so I am not sure if we need to cast the result to `uint32_t`.